macosx/FileNameCell.mm

```diff
@@ -20,45 +20,51 @@
 static CGFloat const kPaddingBetweenNameAndFolderStatus = 4.0;
 static CGFloat const kPaddingExpansionFrame = 2.0;
 
-@interface FileNameCell ()
-
-@property(nonatomic, readonly) NSAttributedString* attributedTitle;
-@property(nonatomic, readonly) NSAttributedString* attributedStatus;
-@property(nonatomic, readonly) NSMutableDictionary* fTitleAttributes;
-@property(nonatomic, readonly) NSMutableDictionary* fStatusAttributes;
-
-@end
-
-@implementation FileNameCell
-
-- (instancetype)init
+static NSParagraphStyle* kParagraphStyle()
 {
-    if ((self = [super init]))
-    {
-        NSMutableParagraphStyle* paragraphStyle = [NSParagraphStyle.defaultParagraphStyle mutableCopy];
-        paragraphStyle.lineBreakMode = NSLineBreakByTruncatingMiddle;
-
-        _fTitleAttributes = [[NSMutableDictionary alloc]
-            initWithObjectsAndKeys:[NSFont messageFontOfSize:12.0], NSFontAttributeName, paragraphStyle, NSParagraphStyleAttributeName, nil];
-
-        NSMutableParagraphStyle* statusParagraphStyle = [NSParagraphStyle.defaultParagraphStyle mutableCopy];
-        statusParagraphStyle.lineBreakMode = NSLineBreakByTruncatingTail;
-
-        _fStatusAttributes = [[NSMutableDictionary alloc]
-            initWithObjectsAndKeys:[NSFont messageFontOfSize:9.0], NSFontAttributeName, statusParagraphStyle, NSParagraphStyleAttributeName, nil];
-    }
-    return self;
+    NSMutableParagraphStyle* paragraphStyle = [NSParagraphStyle.defaultParagraphStyle mutableCopy];
+    paragraphStyle.lineBreakMode = NSLineBreakByTruncatingMiddle;
+    return paragraphStyle;
 }
-
-- (id)copyWithZone:(NSZone*)zone
+static NSParagraphStyle* kStatusParagraphStyle()
 {
-    FileNameCell* copy = [super copyWithZone:zone];
+    NSMutableParagraphStyle* paragraphStyle = [NSParagraphStyle.defaultParagraphStyle mutableCopy];
+    paragraphStyle.lineBreakMode = NSLineBreakByTruncatingTail;
+    return paragraphStyle;
+}
 
-    copy->_fTitleAttributes = _fTitleAttributes;
-    copy->_fStatusAttributes = _fStatusAttributes;
+static NSDictionary<NSAttributedStringKey, id>* const kTitleAttributes = @{
+    NSFontAttributeName : [NSFont messageFontOfSize:12.0],
+    NSParagraphStyleAttributeName : kParagraphStyle(),
+    NSForegroundColorAttributeName : NSColor.controlTextColor
+};
+static NSDictionary<NSAttributedStringKey, id>* const kStatusAttributes = @{
+    NSFontAttributeName : [NSFont messageFontOfSize:9.0],
+    NSParagraphStyleAttributeName : kStatusParagraphStyle(),
+    NSForegroundColorAttributeName : NSColor.secondaryLabelColor
+};
+static NSDictionary<NSAttributedStringKey, id>* const kTitleEmphasizedAttributes = @{
+    NSFontAttributeName : [NSFont messageFontOfSize:12.0],
+    NSParagraphStyleAttributeName : kParagraphStyle(),
+    NSForegroundColorAttributeName : NSColor.whiteColor
```

Comment:
Here and below, does hardcoding `NSColor.whiteColor` play nicely with both light & dark mode?

Reply:
I'm not changing any existing UI: but yes, white is fine here for both light (blue background) & dark (grey background) modes.